### DISCUSSION TITLE
vs 2022 17.1.4 error

### DISCUSSION THREAD

**Discussion by tonyliving** [community/community#14833] [Copilot Conversations]
GitHub Copilot 1.12.0.0 (v1.12.0.0@0660f7705)
GitHub Copilot Agent v1.12.0
[Error] An error occurred for 'CreateAsync': System.ArgumentException: No EditorOptionDefinition export found for the given option name: IntelliCodeGrayTextEnabledOption
参数名: optionId
   在 Microsoft.VisualStudio.Text.EditorOptions.Implementation.EditorOptionsFactoryService.GetOptionDefinitionOrThrow(String optionId)
   在 Microsoft.VisualStudio.Text.EditorOptions.Implementation.EditorOptions.SetOptionValue(String optionId, Object value)
   在 Microsoft.VisualStudio.Internal.InlineSuggestions.Implementation.LegacySuggestionProvider.<SetEnabledOptionAsync>d__30.MoveNext() 位置 D:\a\_work\1\s\src\InlineSuggestions\Implementation\LegacySuggestionProvider.cs:行号 155
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.VisualStudio.Internal.InlineSuggestions.Implementation.LegacySuggestionProvider.<DisabledAsync>d__13.MoveNext() 位置 D:\a\_work\1\s\src\InlineSuggestions\Implementation\LegacySuggestionProvider.cs:行号 66
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.VisualStudio.Internal.InlineSuggestions.Implementation.SuggestionManager.<DisableProviderAsync>d__55.MoveNext() 位置 D:\a\_work\1\s\src\InlineSuggestions\Implementation\SuggestionManager.cs:行号 285
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 GitHub.Copilot.SuggestionManagerAdapter.<DisableProviderAsync>d__3.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 GitHub.Copilot.DisableIntellicodeHandler.<CreateAsync>d__4.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 GitHub.Copilot.CompositeSuggestionHandler.<InitializeAsync>d__11.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 GitHub.Copilot.CopilotSuggestionProvider.<CreateAsync>d__10.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 GitHub.Copilot.CopilotCommandHandler.<>c__DisplayClass34_0.<<GetProvider>b__2>d.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   在 Microsoft.VisualStudio.Threading.JoinableTask`1.CompleteOnCurrentThread()
   在 GitHub.Copilot.CopilotCommandHandler.<>c__DisplayClass34_0.<GetProvider>b__0()
   在 Microsoft.VisualStudio.Utilities.PropertyCollection.GetOrCreateSingletonProperty[T](Object key, Func`1 creator)
   在 GitHub.Copilot.CopilotCommandHandler.GetProvider(ITextView textView, ITextBuffer subjectBuffer)

**Comment by kzu** [comment]
Could you share your Help | About (Copy Info) from VS?

**Reply by tonyliving** [reply]
Microsoft Visual Studio Enterprise 2022
版本 17.1.4
VisualStudio.17.Release/17.1.4+32407.343
Microsoft .NET Framework
版本 4.8.04161

已安装的版本: Enterprise

体系结构关系图和分析工具   00482-10000-00261-AA831
Microsoft 体系结构关系图和分析工具

ASP.NET and Web Tools 2019   17.1.358.51495
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   17.1.358.51495
更多信息，请访问 https://www.asp.net/

Azure 应用服务工具 3.0.0 版   17.1.358.51495
Azure 应用服务工具 3.0.0 版

C# 工具   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
IDE 中使用的 C# 组件。可能使用其他版本的编译器，具体取决于你的项目类型和设置。

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft 库管理器   2.1.161+abc97ecc7d.RR
将客户端库方便地安装到任何 web 项目

NuGet 包管理器   6.1.0
Visual Studio 中的 NuGet 包管理器。有关 NuGet 的详细信息，请访问 https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2206201+62a2c1d6162f828801565a7ca26d9d48b810a05b
提供 ASP.NET Core Razor 的语言服务。

TypeScript Tools   17.0.1229.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic 工具   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
IDE 中使用的 Visual Basic 组件。可能使用其他版本的编译器，具体取决于你的项目类型和设置。

Visual F# Tools   17.1.0-beta.21610.4+07b5673e4f2fa7630e78abe37f16b372353a7242
Microsoft Visual F# Tools

Visual Studio Code 调试适配器主机包   1.0
用于在 Visual Studio 中托管 Visual Studio Code 调试适配器的互操作层

Visual Studio IntelliCode   2.2
Visual Studio 的 AI 协助开发。

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio 容器工具扩展   1.0
查看、管理和诊断 Visual Studio 中的容器。

用于容器的 Microsoft Visual Studio Tools   1.2
在目标环境中开发、运行、验证 ASP.NET Core 应用程序。按 F5，应用程序经调试后直接进入容器，或者按 Ctrl+F5 可编辑和刷新应用，无需重新生成容器。

**Reply by kzu** [reply]
The latest version has improved detection of when the IntelliCode feature is disabled (if installed). This might be what's causing the error in your case. Could you try updating and seeing if you get an improved experience? (Copilot might still not provide suggestions since it needs IntelliCode, but it shouldn't fail).

**Reply by tonyliving** [reply]
Thanks ,I try to enable IntelliCode.It work well
